### PR TITLE
Use `JsonSerializer` to serialize exception data

### PR DESCRIPTION
**Pull request by schmittjoseph**
###### Summary

Switch over to using `JsonSerializer` to serialize our exception data instead of manually serializing. This will make it less likely for us to accidentally introduce bugs here, standardize with our other features, and also make writing more complex json tests for `/exceptions` easier.

**This PR does not change the API surface**.

### CODE REVIEW

File: src/Tools/dotnet-monitor/Exceptions/ExceptionsOperation.cs [hunk 1]
```diff
@@ -141,81 +139,28 @@ private async Task WriteJsonInstance(Stream stream, IExceptionInstance instance,
...
-                    writer.WriteStartArray("frames");
-
-                    StringBuilder builder = new StringBuilder();
-
-                    foreach (var frame in instance.CallStack.Frames)
-                    {
-                        writer.WriteStartObject();
-
-                        string assembledMethodName = frame.MethodName;
-                        if (frame.FullGenericArgTypes.Count > 0)
```
Comment (schmittjoseph) [inline src/Tools/dotnet-monitor/Exceptions/ExceptionsOperation.cs:189]: It looks like `/stacks` and `/exceptions` diverge here. Exception frames include generic arg types while stacks don't... I'll work on restoring this behavior for `/exceptions`

File: src/Microsoft.Diagnostics.Monitoring.WebApi/Models/CallStackResults.cs [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
+        public string MethodNameWithGenericArgTypes
+        {
+            get
+            {
+                StringBuilder builder = new(MethodName);
+                NameFormatter.BuildGenericArgTypes(builder, FullGenericArgTypes);
+                return builder.ToString();
+            }
+            // Only intended for use test code.
+            set
```
Comment (wiktork) [inline src/Microsoft.Diagnostics.Monitoring.WebApi/Models/CallStackResults.cs:24]: internal set

Comment (schmittjoseph) [reply src/Microsoft.Diagnostics.Monitoring.WebApi/Models/CallStackResults.cs:24]: When it's `internal`, `JsonSerializer` is unable to set this field in our tests when we deserialize our api output

File: src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/StackUtilities.cs [hunk 3]
```diff
@@ -17,23 +17,22 @@ internal enum StackFormat
 
     internal static class StackUtilities
     {
-        public static Models.CallStack TranslateCallStackToModel(CallStack stack, NameCache cache, bool methodNameIncludesGenericParameters = true)
+        public static Models.CallStack TranslateCallStackToModel(CallStack stack, NameCache cache, bool parameterTypesSupported = true)
```
Comment (jander-msft) [inline src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/StackUtilities.cs:20]: Can we change the parameter to indicate what it's actually doing? Something like `ensureParameterTypeFieldsNotNull`. I don't know what effect `parameterTypesSupported` is supposed to have without reading the method body.

File: src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/StackUtilities.cs [hunk 4]
```diff
@@ -17,23 +17,22 @@ internal enum StackFormat
...
             stackModel.ThreadId = stack.ThreadId;
             stackModel.ThreadName = stack.ThreadName;
 
-            StringBuilder builder = new();
             foreach (CallStackFrame frame in stack.Frames)
             {
                 var frameModel = CreateFrameModel(frame, cache);
-                if (methodNameIncludesGenericParameters)
+
+                if (parameterTypesSupported)
```
Comment (jander-msft) [inline src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/StackUtilities.cs:30]: Consider adding this as a parameter on `CreateFrameModel` and let it handle all the semantics of what should be included in the model.

**Review by jander-msft — APPROVED** [review]

**Pull request merged**